clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -296,9 +281,11 @@ public boolean awaitPendingRequests(Node node, long timeoutMs) {
      */
     public int pendingRequestCount(Node node) {
         synchronized (this) {
-            List<ClientRequest> pending = unsent.get(node);
-            int unsentCount = pending == null ? 0 : pending.size();
-            return unsentCount + client.inFlightRequestCount(node.idString());
+            int total = 0;
+            for (UnsentRequest request : unsent)
```

Comment:
@hachikuji Are you suggesting that we can add a method `int hasPendingRequests(Node node)` so that user can call this method instead of `pendingRequestCount(node)` if all they need to know is whether there is pending requests for a given node? I am not sure if it worth doing. Strictly speaking the upper bound of time complexity of `hasPendingRequests(node)` will still be `O(n)`, which is the same as `pendingRequestCount(node)`. And adding this method means we have to add more unit tests to cover this method.